pkg/cmd/server/server.go

```diff
@@ -43,12 +45,23 @@ func main() {
 
 	//TODO: for development purposes only. will be remove later
 	fmt.Println("start the ovsdb-etcd server with the following arguments:")
-	fmt.Printf("tcpAddress:%s\nunixAddressress:%s\netcdMembersress:%s\nschemaBasedir:%s\nmaxTasks:%d\ndatabasePrefix:%s\nschemaFile:%s\nschemaName:%s\nloadServerData:%v\n",
+	fmt.Printf("\ttcpAddress:%s\n\tunixAddressress:%s\n\tetcdMembersress:%s\n\tschemaBasedir:%s\n\tmaxTasks:%d\n\tdatabasePrefix:%s\n\tschemaFile:%s\n\tschemaName:%s\n\tloadServerData:%v\n",
 		*tcpAddress, *unixAddress, *etcdMembers, *schemaBasedir, *maxTasks, *databasePrefix, *schemaFile, *schemaName, *loadServerDataFlag)
 
 	if len(*tcpAddress) == 0 && len(*unixAddress) == 0 {
 		klog.Fatal("You must provide a network-address (TCP and/or UNIX) to listen on")
 	}
+	// several OVSDB ceployments can share the same etcd, but for rest of the work, we don't have to separate
+	// databasePrefix and serviceName.
+	if len(*databasePrefix) != 0 {
+		if len(*serviceName) != 0 {
+			common.SetPrefix(*databasePrefix + common.KEY_DELIMETER + *serviceName)
```

Comment:
is this setting a global variable? if so that rather we don't

Reply:
Yes it is a global variable and it is set during the server startup (one of the common patterns like logging), otherwise, we have to have a pointer to some commonplace with this configuration in each place.